library/universes.ml

```diff
@@ -756,54 +793,78 @@ let minimize_univ_variables ctx us algs left right cstrs =
 	let lbounds' = 
 	  match compute_lbound (List.map (fun (d,l) -> d, Universe.make l) lower) with
 	  | None -> lbounds
-	  | Some lbound -> LMap.add r (true, false, lbound) lbounds
+	  | Some lbound -> LMap.add r (true, false, lbound, lower_of_list lower)
+                                   lbounds
 	in (Univ.LMap.remove r left, lbounds'))
       left (left, Univ.LMap.empty)
   in
   let rec instance (ctx', us, algs, insts, cstrs as acc) u =
-    let acc, left = 
+    let acc, left, lower =
       try let l = LMap.find u left in
-	    List.fold_left
-	      (fun (acc, left') (d, l) -> 
-	       let acc', (enf,alg,l') = aux acc l in
-	       let l' = 
-		 if enf then Universe.make l 
-		 else l' 
-	       in acc', (d, l') :: left')
-	      (acc, []) l
-      with Not_found -> acc, []
+	  let acc, left, newlow, lower =
+            List.fold_left
+	      (fun (acc, left', newlow, lower') (d, l) ->
+	        let acc', (enf,alg,l',lower) = aux acc l in
+	        let l' =
+		  if enf then Universe.make l
+		  else l'
+	        in acc', (d, l') :: left',
+                   lower_add l d newlow, lower_union lower lower')
```

Comment:
Could there be a conflict when doing the union here? (I am always wary of map unions.)